Aspire Budgeting/Facilities/AppCoordinator.swift

```diff
@@ -0,0 +1,69 @@
+//
+// AppCoordinator.swift
+// Aspire Budgeting
+//
+
+import Foundation
+import Combine
+
+final class AppCoordinator: ObservableObject {
+  private let stateManager: AppStateManager
+  private let localAuthorizer: AppLocalAuthorizer
+
+  private var stateManagerSink: AnyCancellable!
+
+  init(stateManager: AppStateManager,
+       localAuthorizer: AppLocalAuthorizer) {
+    self.stateManager = stateManager
+    self.localAuthorizer = localAuthorizer
+  }
+
+  func start() {
+    stateManagerSink = stateManager
+      .currentState
+      .receive(on: DispatchQueue.main)
+      .sink {
+        self.objectWillChange.send()
+        self.handle(state: $0)
+      }
+  }
+
+  func pause() {
+    self.stateManager.handleBackground()
+  }
+
+  func resume() {
+    if needsLocalAuth {
+      self.localAuthorizer.authenticateUserLocally {self.stateManager.authenticatedLocally(result: $0)}
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/0064a2de7513bf35f3612cdd6552d081d2b41cdd/Aspire Budgeting/Facilities/AppCoordinator.swift#L37"></span>Closure expressions should have a single space inside each brace.<br />
<code>closure_spacing</code> <code>AppCoordinator.swift:37</code></td>
    </tr>
  </tbody>
</table>